Languages/STATA/README.md

```diff
@@ -18,142 +67,223 @@ describe
- size:         3,478                          (_dta has notes)
+Contains data from /n/sw/stata-17/ado/base/a/auto.dta
+ Observations:            74                  1978 automobile data
+    Variables:            13                  13 Apr 2020 17:45
+                                              (_dta has notes)
 -------------------------------------------------------------------------------
-              storage   display    value
-variable name   type    format     label      variable label
+Variable      Storage   Display    Value
+    name         type    format    label      Variable label
 -------------------------------------------------------------------------------
-make            str18   %-18s                 Make and Model
+make            str18   %-18s                 Make and model
 price           int     %8.0gc                Price
 mpg             int     %8.0g                 Mileage (mpg)
-rep78           int     %8.0g                 Repair Record 1978
+rep78           int     %8.0g                 Repair record 1978
 headroom        float   %6.1f                 Headroom (in.)
 trunk           int     %8.0g                 Trunk space (cu. ft.)
 weight          int     %8.0gc                Weight (lbs.)
 length          int     %8.0g                 Length (in.)
-turn            int     %8.0g                 Turn Circle (ft.)
+turn            int     %8.0g                 Turn circle (ft.)
 displacement    int     %8.0g                 Displacement (cu. in.)
-gear_ratio      float   %6.2f                 Gear Ratio
-foreign         byte    %8.0g      origin     Car type
-price2          float   %9.0g                 
+gear_ratio      float   %6.2f                 Gear ratio
+foreign         byte    %8.0g      origin     Car origin
+price2          float   %9.0g
 -------------------------------------------------------------------------------
-Sorted by:  foreign
-     Note:  dataset has changed since last saved
+Sorted by: foreign
+     Note: Dataset has changed since last saved.
 
 . exit
 
 end of do-file
 ```
 
-#### REFERENCES:
+## Multi-core job
+
+This example shows how to run a Stata do file in multi-core mode (or
+multiprocessor, mp).
+
+Example STATA do file
+
+Note that you will have to change the number of processor in thei first line of
+do file to match how many cores you request on the `run_mp.sbatch` file.
+
+```
+set processors 4
+display "Hello, World!"
+```
+
+Batch-job submission script
+
+```bash
+#!/bin/bash
+#SBATCH -J my_stata_job       # job name
+#SBATCH -o my_stata_job.out   # standard output file
+#SBATCH -e my_stata_job.err   # standard error file
+#SBATCH -p serial_requeue     # partition
+#SBATCH -t 0-00:30            # time in D-HH:MM
+#SBATCH -N 1                  # number of nodes
+#SBATCH -c 4                  # number of cores
+#SBATCH --mem=4000            # total memory in MB
+
+# Load required modules
+module load stata/17.0-fasrc01
+
+# Run program
+stata-mp -b hello_mp.do
```

Comment:
This should be prefixed by srun per: https://docs.rc.fas.harvard.edu/kb/running-jobs/#Using_Threads_such_as_OpenMP